openmc/mgxs/mgxs.py

```diff
@@ -4615,3 +4628,728 @@ def get_pandas_dataframe(self, groups='all', nuclides='all',
+        cross section data data for one or more energy groups and subdomains.
+
+        Parameters
+        ----------
+        groups : Iterable of Integral or 'all'
+            Energy groups of interest. Defaults to 'all'.
+        subdomains : Iterable of Integral or 'all'
+            Subdomain IDs of interest. Defaults to 'all'.
+        nuclides : Iterable of str or 'all' or 'sum'
+            A list of nuclide name strings (e.g., ['U-235', 'U-238']). The
+            special string 'all' will return the cross sections for all nuclides
+            in the spatial domain. The special string 'sum' will return the
+            cross section summed over all nuclides. Defaults to 'all'.
+        xs_type: {'macro', 'micro'}
+            This parameter is not relevant for chi but is included here to
+            mirror the parent MGXS.get_xs(...) class method
+        order_groups: {'increasing', 'decreasing'}
+            Return the cross section indexed according to increasing or
+            decreasing energy groups (decreasing or increasing energies).
+            Defaults to 'increasing'.
+        value : {'mean', 'std_dev', 'rel_err'}
+            A string for the type of value to return. Defaults to 'mean'.
+
+        Returns
+        -------
+        numpy.ndarray
+            A NumPy array of the multi-group cross section indexed in the order
+            each group, subdomain and nuclide is listed in the parameters.
+
+        Raises
+        ------
+        ValueError
+            When this method is called before the multi-group cross section is
+            computed from tally data.
+
+        """
+
+        cv.check_value('value', value, ['mean', 'std_dev', 'rel_err'])
+        cv.check_value('xs_type', xs_type, ['macro', 'micro'])
+
+        filters = []
+        filter_bins = []
+
+        # Construct a collection of the domain filter bins
+        if not isinstance(subdomains, basestring):
+            cv.check_iterable_type('subdomains', subdomains, Integral, max_depth=2)
+            for subdomain in subdomains:
+                filters.append(self.domain_type)
+                filter_bins.append((subdomain,))
+
+        # Construct list of energy group bounds tuples for all requested groups
+        if not isinstance(groups, basestring):
+            cv.check_iterable_type('groups', groups, Integral)
+            for group in groups:
+                filters.append('energyout')
+                filter_bins.append((self.energy_groups.get_group_bounds(group),))
+
+        # If chi prompt was computed for each nuclide in the domain
+        if self.by_nuclide:
+
+            # Get the sum as the fission source weighted average chi for all
+            # nuclides in the domain
+            if nuclides == 'sum' or nuclides == ['sum']:
+
+                # Retrieve the fission production tallies
+                prompt_nu_fission_in = self.tallies['nu-fission-in'] - \
```

Comment:
Just curious, do you think it is worth storing `prompt_nu_fission_in` and `prompt_nu_fission_out` in the `tallies` attribute when the cross section is first computed? This would avoid redundant calculations in the event that methods like `get_xs(...)` are called a bunch of times (which can be a common use case in my experience).